plugins/azure/resoto_plugin_azure/resource/network.py

```diff
@@ -3969,6 +4257,48 @@ class AzureNetworkProfile(AzureResource):
     etag: Optional[str] = field(default=None, metadata={'description': 'A unique read-only string that changes whenever the resource is updated.'})  # fmt: skip
     provisioning_state: Optional[str] = field(default=None, metadata={'description': 'The current provisioning state.'})  # fmt: skip
     resource_guid: Optional[str] = field(default=None, metadata={'description': 'The resource GUID property of the network profile resource.'})  # fmt: skip
+    _network_interfaces_and_vm_ids: Optional[List[Tuple[str, str]]] = None
+
+    def post_process(self, graph_builder: GraphBuilder, source: Json) -> None:
+        def collect_nis_and_vms() -> None:
+            resources_api_spec = AzureApiSpec(
+                service="network",
+                version="2023-05-01",
+                path="/subscriptions/{subscriptionId}/providers/Microsoft.Network/networkInterfaces",
+                path_parameters=["subscriptionId"],
+                query_parameters=["api-version"],
+                access_path="value",
+                expect_array=True,
+            )
+
+            self._network_interfaces_and_vm_ids = [
+                (r["id"], r["properties"]["virtualMachine"]["id"])
+                for r in graph_builder.client.list(resources_api_spec)
+                if "properties" in r and "virtualMachine" in r["properties"]
+            ]
+
+        graph_builder.submit_work(collect_nis_and_vms)
```

Comment:
To connect `AzureNetworkProfile` between `AzureVirtualMachine`, we need to know `AzureNetworkInterface`. Because `NetworkInterface` contains `ID` of `VM` and we have access only to the `ID` of `NetworkInterface` in `AzureNetworkProfile`.